File: output/golang/models/struct/LogEntry.go
package main

type LogEntry struct {
	DeviceState DeviceState `json:"device_state" bson:"device_state"`
	TimeFired *int `json:"time_fired" bson:"time_fired"`
	Values map[string]string `json:"values" bson:"values"`
	Tag *string `json:"tag" bson:"tag"`
	StackTrace *string `json:"stack_trace" bson:"stack_trace"`
	Type *string `json:"type" bson:"type"`
	Message *string `json:"message" bson:"message"`
}

File: output/kotlin/models/overridden/LogEntry.kt
package `overridden`

interface ILogEntry {
    val deviceState: DeviceState
    val timeFired: Long
    val values: Map<String, String>
    val tag: String
    val stackTrace: String
    val type: String
    val message: String
}

data class LogEntry(
    override val deviceState: DeviceState,
    override val timeFired: Long,
    override val values: Map<String, String>,
    override val tag: String,
    override val stackTrace: String,
    override val type: String,
    override val message: String,
) : ILogEntry